eng/native/init-vs-env.cmd

```diff
@@ -0,0 +1,75 @@
+@if not defined _echo @echo off
+
+:: Initializes Visual Studio developer environment. If a build architecture is passed
+:: as an argument, it also initializes VC++ build environment and CMakePath.
+
+set "__VCBuildArch="
+if /i "%~1" == "x86"   (set __VCBuildArch=x86)
+if /i "%~1" == "x64"   (set __VCBuildArch=x86_amd64)
+if /i "%~1" == "arm"   (set __VCBuildArch=x86_arm)
+if /i "%~1" == "arm64" (set __VCBuildArch=x86_arm64)
+if /i "%~1" == "wasm"  (set __VCBuildArch=x86_amd64)
+
+:: Default to highest Visual Studio version available that has Visual C++ tools.
+::
+:: For VS2017 and later, multiple instances can be installed on the same box SxS and VS1*0COMNTOOLS
+:: is no longer set as a global environment variable and is instead only set if the user
+:: has launched the Visual Studio Developer Command Prompt.
+::
+:: Following this logic, we will default to the Visual Studio toolset assocated with the active
+:: Developer Command Prompt. Otherwise, we will query VSWhere to locate the later version of
+:: Visual Studio available on the machine. Finally, we will fail the script if no supported
+:: instance can be found.
+
+if defined VisualStudioVersion goto :VSDetected
+
+set "__VSWhere=%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe"
+set "__VSCOMNTOOLS="
+
+if exist "%__VSWhere%" (
+    for /f "tokens=*" %%p in (
+        '"%__VSWhere%" -latest -prerelease -products * -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath'
+    ) do set __VSCOMNTOOLS=%%p\Common7\Tools
+)
+
+if not exist "%__VSCOMNTOOLS%" goto :VSMissing
+
+:: Make sure the current directory stays intact
+set "VSCMD_START_DIR=%CD%"
+
+call "%__VSCOMNTOOLS%\VsDevCmd.bat" -no_logo
+
+:: Clean up helper variables
+set "__VSWhere="
+set "__VSCOMNTOOLS="
+set "VSCMD_START_DIR="
+
+:VSDetected
+if "%VisualStudioVersion%"=="16.0" (
+    set __VSVersion=vs2019
+    set __PlatformToolset=v142
+    goto :SetVCEnvironment
+) else if "%VisualStudioVersion%"=="15.0" (
+    set __VSVersion=vs2017
```

Comment:
We do not support building with VS2017 anymore. You can delete this part.

Reply:
Also removed two remaining DIA SDK checks in the two scripts that call `init-vs-env.cmd` without passing an architecture and later optionally run `vcvarsall.bat %__VCBuildArch%`. With the current changes, successful execution of `init-vs-env.cmd` implies we must have DIA SDK. If we decide to relax that requirement for building managed components only, we will have to put DIA SDK checks or equivalent VC Tools checks back.